File: minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: TreeNode) -> int:
        
        if not root:
            return 0
        queue = [root]
        level = 1
        
        while queue:
            
            len_q = len(queue)
            
            for i in range(len_q):
                
                curr_root = queue.pop(0)
                
                if not curr_root.left and not curr_root.right:
                    return level
                if curr_root.left:
                    queue.append(curr_root.left)
                if curr_root.right:
                    queue.append(curr_root.right)
                
                
            level+=1
        
        return level
        
        
        

File: minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        
        if (root == null) return 0;
        
        Queue<TreeNode> q = new LinkedList<>();
        
        q.add(root);
    
        int i = 0;
        int level = 1;
        while(!q.isEmpty()){
            int len_q = q.size();
            for(int j = 0; j < len_q; j++){
                TreeNode currRoot = q.remove();
                
                if(currRoot.left == null && currRoot.right == null){
                   return level;
                }
                if(currRoot.left != null)q.add(currRoot.left);
                if(currRoot.right != null)q.add(currRoot.right);
            }
            level++;
            
        }
        return 0;
    }
    
    

    
}